### ISSUE TITLE
Error during install.sh execution - Lambda function is too large

### ISSUE THREAD

**Issue by medhost-jshaver** [awslabs/fhir-works-on-aws-deployment#166]
Cloned from mainline and while following these steps below I ran into an error during serverless deploy:
https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/INSTALL.md#linux-or-osx-installation

Here is the error

>  Serverless Error ---------------------------------------
>  
>   An error occurred: UpdateStatusLambdaFunction - Unzipped size must be smaller than 262144000 bytes (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 3632f87a-4d09-45e7-8be8-5b5babd30a03; Proxy: null).
>  
>   Get Support --------------------------------------------
>      Docs:          docs.serverless.com
>      Bugs:          github.com/serverless/serverless/issues
>      Issues:        forum.serverless.com
>  
>   Your Environment Information ---------------------------
>      Operating System:          darwin
>      Node Version:              15.0.1
>      Framework Version:         2.12.0 (local)
>      Plugin Version:            4.1.2
>      SDK Version:               2.3.2
>      Components Version:        3.4.0
 

Looking at the fhir-service.zip file that is packaged during this process, it appears to have roughly 540MB in node_modules when uncompressed. 
```
┌─[~/Downloads/fhir-service] - [2020-12-08 12:04:20]
└─[0] <> du -sh *
 24K	bulkExport
8.0K	ddbToEsLambda
540M	node_modules
4.0K	package.json
 24K	src
```

**Comment by rsmayda** [comment]
Hmm that is really weird.. What command did you run to get this output? Could you delete your node_modules folder and try again?

**Comment by medhost-jshaver** [comment]
Here is what I ran -- my best attempt at a sanity check

1. mkdir {newDirectory}
2. cd {newDirecotry}
3. git clone https://github.com/awslabs/fhir-works-on-aws-deployment.git
4. chmod +x ./scripts/install.sh
5. sudo ./scripts/install.sh
6. sudo -E ./scripts/install.sh --stage dev-marlin-1 --region us-east-1




after deleting the node_modules folder and re running I get the same file size. 

Here are some of the larger files I see in node_modules:
```
┌─[~/Downloads/fhir-service/node_modules] - [2020-12-09 09:35:53]
└─[0] <> du -sh * | sort -h 
 12K	buffer-alloc
 12K	buffer-alloc-unsafe

... skipping to bottom ....

8.0M	json-refs
8.2M	@types
 12M	fhir-works-on-aws-routing
 17M	rxjs
 27M	@serverless
 30M	@babel
 49M	dependency-tree
 49M	fhir-works-on-aws-search-es
 49M	filing-cabinet
 50M	fhir-works-on-aws-persistence-ddb
 55M	aws-sdk
```

I went to dig up an old fhir-service package from s3 that I have deployed and I see a pretty significant difference in the fhir-works package size.  The node_modules size is quite large even in the older package (298M), but it still deployed the functions. Not 100% sure how that works but I assume it doesn't use the whole node_modules directory considering that the limit is for a lambda is 262M.

Here is the top level of the "old" (deployed Nov 5th) fhir-service package
```
┌─[~/Downloads/fhir-service-old] - [2020-12-09 09:52:51]
└─[0] <> du -sh *
8.0K	ddbToEsLambda
298M	node_modules
4.0K	package.json
 36K	src
```
And here are some of the larger files within node_modules
```
┌─[~/Downloads/fhir-service-old/node_modules] - [2020-12-09 09:53:36]
└─[0] <> du -sh * | sort -h

... skipping to bottom ...

4.2M	protobufjs
4.8M	lodash
5.0M	moment
5.3M	@babel
5.3M	fhir-works-on-aws-search-es
5.5M	node-notifier
7.5M	fhir-works-on-aws-routing
8.2M	@types
 11M	@serverless
 17M	rxjs
 48M	aws-sdk
 53M	detective-typescript
```

I am curious though, are you seeing different results for the package size?

**Comment by rsmayda** [comment]
Doing a quick look into my S3 deployment bucket I am seeing the .zip file being ~11 MB and the unzipped folder being ~55 MB. Let me talk with some teammates to figure out if we can replicate this.

**Comment by Bingjiling** [comment]
I did a brand new deployment and got the .zip file being 33MB and the unzipped folder being 168MB. Here're the largest files under node_modules: 
<img width="618" alt="Screen Shot 2020-12-09 at 2 03 19 PM" src="https://user-images.githubusercontent.com/14276357/101675110-61919b00-3a27-11eb-8d91-bd017e99ef99.png">

What operating system did you use to deploy? I would recommend try [install using Docker](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/INSTALL.md#docker-installation) and see if that resolves the issue. Hope this helps.

**Comment by medhost-jshaver** [comment]
I am using macOS Catalina v10.15.7

I started looking into what could be causing so much bloat in the node modules and after running a number of commands I got to a point where my fhir service package looks more reasonable. 

I think I caused this issue by running some of those local development deployment scripts in fhir-works-on-aws-deployment/scripts. Perhaps npm retained some of the dev dependencies somehow.

Some commands that I believe helped the most. (last command was to clear global installs)
- sudo npm prune
- sudo npm dedupe
- sudo npm list -g --depth=0. | awk -F ' ' '{print $2}' | awk -F '@' '{print $1}'  | sudo xargs npm remove -g

I also deleted the node_modules folder again (not sure if this is redundant)

After these commands I re ran install.sh and this is what node modules looks like for me after that. It is strange how small the fhir-works-on-aws packages became. 

I think we should be good on closing this ticket, and I may try that docker container out and see how it compares as well. Thanks for the help @rsmayda and @Bingjiling!

![image](https://user-images.githubusercontent.com/74723324/101681491-d238b780-3a27-11eb-8d1c-3db69bcab955.png)
